help_function/src/cub_blockexchange_tests.cpp

```diff
@@ -0,0 +1,148 @@
+// ====------ cub_blockexchange_tests.cpp---------- -*- C++ -* ----===////
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//
+// ===----------------------------------------------------------------------===//
+#include <sycl/sycl.hpp>
+#include <dpct/dpct.hpp>
+#include <dpct/dpl_extras/dpcpp_extensions.h>
+#include <sycl/ext/oneapi/group_local_memory.hpp>
+
+#include <stdio.h>
+#include <string.h>
+
+#define VALUES_PER_THREAD  4
+#define SIZE  16
+
+void init_data(int* data, int num) {
+  for(int i = 0; i < num; i++)
+    data[i] = i;
+}
+void verify_data(int* data, int num) {
+  return;
+}
+void print_data(int* data, int num) {
+  for (int i = 0; i < num; i++) {
+    std::cout << data[i] << " ";
+  }
+  std::cout << std::endl;
+}
+
+void striped_to_blocked_kernel(int* data,
+                               sycl::nd_item<3> item_ct1){
+                            
+                            
+  int threadid = item_ct1.get_local_id(2);
+  uint8_t* _local_memory = sycl::ext::oneapi::group_local_memory_for_overwrite<uint8_t*>(item_ct1.get_group());
+  dpct::group::exchange<data, VALUES_PER_THREAD> exchange(_local_memory).striped_to_blocked(item_ct1.get_group(), data);
+  item_ct1.barrier(sycl::access::fence_space::local_space);
+                            
+}
+
+void blocked_to_striped_kernel(int* data,
+                               sycl::nd_item<3> item_ct1){
+                        
+  int threadid = item_ct1.get_local_id(2);
+  uint8_t* _local_memory = sycl::ext::oneapi::group_local_memory_for_overwrite<uint8_t*>(item_ct1.get_group());
+  dpct::group::exchange<data, VALUES_PER_THREAD> exchange(_local_memory).blocked_to_striped(item_ct1.get_group(), data);

```

Comment:
We can't pass a lvalue as class template argument.
```suggestion
  dpct::group::exchange<int, VALUES_PER_THREAD> exchange(_local_memory).blocked_to_striped(item_ct1.get_group(), data);

```